### ISSUE TITLE
GCM token NotRegistered

### ISSUE THREAD

**Issue by alextomas** [phonegap-build/PushPlugin#466]
I have a problem with notifications GCM and tokens expiry

Check the token device and sending a notification, but over time the token is no longer valid and get an error: NotRegistered. I saw that my application registers a different register_id every time I run the application.

What is happening?

Thank You!

**Comment by andreszs** [comment]
Try to debug your server side to see if you are sending the POST request exactly as it should be sent, or maybe you are improperly reading/sending/storing it on your server.

Also, remember that every time you reinstall the app, the previous token will become obsolete, you must get a new token/regid every time the app is installed/reinstalled.

**Comment by cyriljoui** [comment]
Hi !

Where did you get this information?
Could you please give us the link (Google Developer?).
I think there is something missing on PushPlugin (Android but maybe also in IOS ...), every time we must call register() on plugin (even if application is already registered).

We may implement register execution like this
				// Already registered?
				String regId = GCMRegistrar.getRegistrationId(getApplicationContext());
				if (regId == null) {
					Log.d(TAG, "New registration needed");
					GCMRegistrar.register(getApplicationContext(), gSenderID);
				} else {
					Log.d(TAG, "Already registered! go on directly ...");
					callbackContext.success();
					// Result (create this method) see after
					sendJavascriptRegistered(regId);
				}


	public static void sendJavascriptRegistered(String regId) {
		JSONObject json;

		try
		{
			json = new JSONObject().put("event", "registered");
			json.put("regid", regId);

			Log.v(TAG, "onRegistered: " + json.toString());

			// Send this JSON data to the JavaScript application above EVENT should be set to the msg type
			// In this case this is the registration ID
			sendJavascript( json );

		}
		catch( JSONException e)
		{
			// No message to the user is sent, JSON failed
			Log.e(TAG, "onRegistered: JSON exception");
		}		
	}

**Comment by ghominejad** [comment]
I think the problem occurs when you run the register function with no internet access then in the next runs (one or two) you'll get new registration id!

**Comment by ghominejad** [comment]
In the most of times the problem occurs when you run the register function with no internet access then in the next runs (one or two) you'll get new registration id!
I've solved the problem by creating new `bindEvents` function. i call this function when i haven't internet access otherwise i call the register.

**Comment by cyriljoui** [comment]
I've never got any issue with getRegistrationId() ... I will check with Google about expiration.
Thank you for your feedback about no internet access, you're right ! Did you push back your fix to this repository?

**Comment by ghominejad** [comment]
@cyriljoui Yes, (#497), it checks network connectivity before calling `GCMRegistrar.register`

**Comment by alexislg2** [comment]
So could you explain exactly what is the fix to this annoying issue?
